handler.go

```diff
@@ -28,11 +28,14 @@ type handler struct {
 
 func newHandler(do reflect.Value) *handler {
 	return &handler{
-		do:                       do,
-		userDefinedContextOffset: -1,
-		cookiesOffset:            -1,
-		headerOffset:             -1,
-		matchedPathIndex:         -1,
+		do:                 do,
+		userContextsOffset: make([]int, 0),
```

Comment:
I think it should be userContextOffsets actually since what we mean is multiple offsets here

Reply:
I thinking is we have multiple user contexts and each context has a single offset 🤔

Reply:
Because we actually store `offsets` here, but it's fine, I think the better way is creating an agent for UserContext to make these be hidden behind it.